File: java/base/src/main/java/vsdk/toolkit/common/dataStructures/CircularListIterator.java
package vsdk.toolkit.common.dataStructures;

public class CircularListIterator<T> extends CircularListBaseIterator {
    public CircularListIterator(CircularList<T> list) {
        super(list.baseList());
    }

    public T nextOnSequence() {
        CircularListNode<T> link = (CircularListNode<T>)next();
        return link != null ? link.data : null;
    }

    public void init(CircularList<T> list) {
        super.init(list.baseList());
    }
}

File: typescript/base/src/main/vsdk/toolkit/common/dataStructures/CircularListIterator.ts
import { CircularList } from "./CircularList";
import { CircularListBaseIterator } from "./CircularListBaseIterator";
import { CircularListNode } from "./CircularListNode";

export class CircularListIterator<T> extends CircularListBaseIterator {
  public constructor(list: CircularList<T>) {
    super(list.baseList());
  }

  public nextOnSequence(): T | null {
    const link = this.next() as CircularListNode<T> | null;
    return link !== null ? link.data : null;
  }

  public override init(list: CircularList<T>): void {
    super.init(list.baseList());
  }
}
